DEVDOCS-5401: [update] add price list webhooks (#148)
bigcommerce/docs: docs/api-docs/webhooks/webhook-events.mdx

Consult the [pages section of the Channel Webhooks Guide](/docs/integrations/webhooks/events/channels#pages).

## Price Lists
The following price lists webhook events fire in response to actions that affect a specific price list:

| Name / Scope | Description |
|:-------------|:------------|
| `store/pricelist/created`  | Fires when a price list is created. | 
| `store/pricelist/updated/` | Fires when a price list is updated. | 
| `store/pricelist/activated/` | Fires when a price list is activated. | 
| `store/pricelist/deactivated/` | Fires when a price list is deactivated. | 
| `store/pricelist/deleted/` | Fires when a price list is deleted. | 

> | `store/pricelists/deleted/` | Fires when price lists are deleted. | 

Reviewer: Would it make sense to add a sample payload below for this hook, as it's a batch webhook?

Author: The batch webhook has the same payload.

* `store/pricelist/created`
* `store/pricelist/updated/` 
* `store/pricelist/activated/`
* `store/pricelist/deactivated/`
* `store/pricelist/deleted/`

```json filename="Example price list created payload object" showLineNumbers copy
{
  "scope": "store/priceList/created",
  "store_id": "1001197568",
  "data": {
    "price_list_id": 5
  },
  "hash": "e0c298b8097a6a2f39d17e593a9b360f5b2fef7d",
  "created_at": 1709327895,
  "producer": "stores/{store_hash}"
}
```

## Price list records

The following price lists records webhook events fire in response to actions that affect a specific price list record:

| Name / Scope | Description |
|:-------------|:------------|
| `store/pricelist/record/created`  | Fires when a price list is created. | 
| `store/pricelist/records/created/` | Fires when price list records are updated. | 
| `store/pricelist/record/updated/` | Fires when a price list record is updated. | 
| `store/pricelist/records/updated/` | Fires when price list records are updated. | 
| `store/pricelist/records/updated/` | Fires when price list records are updated. | 
| `store/pricelist/record/deleted/` | Fires when a price list record is deleted. | 

> | `store/pricelist/records/deleted/` | Fires when price list records are deleted. | 

Reviewer: Similar comment as above for the batch hooks- they all take the same form for price records.

Author: The batch hooks have the same payload. This is what I was able to confirm today.

{
  "scope": "store/priceLists/deleted",
  "store_id": "1001197568",
  "data": {
    "price_list_ids": [
      5
    ]
  },
  "hash": "e0c298b8097a6a2f39d17e593a9b360f5b2fef7d",
  "created_at": 1709327895,
  "producer": "stores/{store_hash}"
}
```

## Price list records

The following price lists records webhook events fire in response to actions that affect a specific price list record:

| Name / Scope | Description |
|:-------------|:------------|
| `store/priceList/record/created`  | Fires when a price list is created. | 
| `store/priceList/records/created/` | Fires when price list records are updated. | 
| `store/priceList/record/updated/` | Fires when a price list record is updated. | 
| `store/priceList/records/updated/` | Fires when price list records are updated. | 
| `store/priceList/record/deleted/` | Fires when a price list record is deleted. | 

Payload objects with the following scopes take the form that follows:

* `store/pricelist/record/created`
* `store/pricelist/records/updated/` 
* `store/pricelist/records/deleted/`

```json filename="Example price list records created payload object" showLineNumbers copy
{

>   "scope": "store/priceList/records/created",

Reviewer: This is the payload for `priceList/record/created`

The one for multiple records looks like:
<img width="527" alt="Screenshot 2024-03-07 at 9 13 31 AM" src="https://github.com/bigcommerce/docs/assets/1259260/4937f256-5d05-42e5-9c98-18fdb72f7ed1">

Consult the [pages section of the Channel Webhooks Guide](/docs/integrations/webhooks/events/channels#pages).

## Price Lists
The following price lists webhook events fire in response to actions that affect a specific price list:

| Name / Scope | Description |
|:-------------|:------------|
| `store/priceList/created`  | Fires when a price list is created. | 
| `store/priceList/updated/` | Fires when a price list is updated. | 
| `store/priceList/activated/` | Fires when a price list is activated. | 
| `store/priceList/deactivated/` | Fires when a price list is deactivated. | 

> | `store/priceLists/deleted/` | Fires when a price list is deleted. | 

Reviewer: There is also a singular store/priceList/deleted/ webhook

Payload objects with the following scopes take the form that follows:

* `store/priceList/created`
* `store/priceList/updated/` 
* `store/priceList/activated/`
* `store/priceList/deactivated/`
* `store/priceLists/deleted/`

```json filename="Example price list created payload object" showLineNumbers copy
{
  "scope": "store/priceList/created",
  "store_id": "1001197568",
  "data": {
    "price_list_id": 5
  },
  "hash": "e0c298b8097a6a2f39d17e593a9b360f5b2fef7d",
  "created_at": 1709327895,
  "producer": "stores/{store_hash}"
}

> ```

Reviewer: Do we want to move the line for 
* `store/priceLists/deleted/`

Down here, over the associated type of json?  I don't have a good sense of how these things look laid out on the page.

```json filename="Example price lists deleted payload object" showLineNumbers copy
{
  "scope": "store/priceLists/deleted",
  "store_id": "1001197568",
  "data": {
    "price_list_ids": [
      5
    ]
  },
  "hash": "e0c298b8097a6a2f39d17e593a9b360f5b2fef7d",
  "created_at": 1709327895,
  "producer": "stores/{store_hash}"
}
```

## Price list records

The following price lists records webhook events fire in response to actions that affect a specific price list record:

| Name / Scope | Description |
|:-------------|:------------|
| `store/priceList/record/created`  | Fires when a price list is created. | 
| `store/priceList/records/created/` | Fires when price list records are updated. | 
| `store/priceList/record/updated/` | Fires when a price list record is updated. | 
| `store/priceList/records/updated/` | Fires when price list records are updated. | 
| `store/priceList/record/deleted/` | Fires when a price list record is deleted. | 

> 

Reviewer: Should there be a json payload for these here?

Author: @megdesko The payload object is on line 625?

Consult the [pages section of the Channel Webhooks Guide](/docs/integrations/webhooks/events/channels#pages).

## Price Lists

> The following price lists webhook events fire in response to actions that affect a specific price list:

Reviewer 2: ```suggestion
The following webhook events fire in response to actions that affect a specific price list:
```

| Name / Scope | Description |
|:-------------|:------------|
| `store/priceList/created`  | Fires when a price list is created. | 
| `store/priceList/updated/` | Fires when a price list is updated. | 
| `store/priceList/activated/` | Fires when a price list is activated. | 
| `store/priceList/deactivated/` | Fires when a price list is deactivated. | 
| `store/priceList/deleted/` | Fires when a price list is deleted. | 

Payload objects with the following scopes take the form that follows:

* `store/priceList/created`
* `store/priceList/updated/` 
* `store/priceList/activated/`
* `store/priceList/deactivated/`
* `store/priceList/deleted/`

> ```json filename="Example price list created payload object" showLineNumbers copy

Reviewer 2: ```suggestion
```json filename="Example payload object for creating a price list" showLineNumbers copy
```

{
  "scope": "store/priceList/created",
  "store_id": "1001197568",
  "data": {
    "price_list_id": 5
  },
  "hash": "e0c298b8097a6a2f39d17e593a9b360f5b2fef7d",
  "created_at": 1709327895,
  "producer": "stores/{store_hash}"
}
```

> The following price lists batch webhook events fire in response to actions that affect price lists:

Reviewer 2: ```suggestion
The following batch webhook events fire in response to actions that affect price lists:
```

| Name / Scope | Description |
|:-------------|:------------|
| `store/priceLists/deleted/` | Fires when price list are deleted. | 

Payload objects with the following scopes take the form that follows:
* `store/priceLists/deleted/`

> ```json filename="Example price lists deleted payload object" showLineNumbers copy

Reviewer 2: ```suggestion
```json filename="Example payload object for deleting price lists" showLineNumbers copy
```

{
  "scope": "store/priceLists/deleted",
  "store_id": "1001197568",
  "data": {
    "price_list_ids": [
      5
    ]
  },
  "hash": "e0c298b8097a6a2f39d17e593a9b360f5b2fef7d",
  "created_at": 1709327895,
  "producer": "stores/{store_hash}"
}
```

## Price list records

> The following price lists records webhook events fire in response to actions that affect a specific price list record:

Reviewer 2: ```suggestion
The following webhook events fire in response to actions that affect a specific price list record:
```

| Name / Scope | Description |
|:-------------|:------------|
| `store/priceList/record/created/` | Fires when a price list record is created. | 
| `store/priceList/record/updated/` | Fires when a price list record is updated. | 
| `store/priceList/record/deleted/` | Fires when a price list record is deleted. | 

Payload objects with the following scopes take the form that follows:

* `store/priceList/record/created`
* `store/priceList/record/updated/` 
* `store/priceList/record/deleted/`

```json filename="Example price list record created payload object" showLineNumbers copy
{
  "scope": "store/priceList/record/created",
  "store_id": "1001197568",
  "data": {
    "price_list_id": 1,
    "variant_id": 179,
    "currency": "USD"
      },
  "hash": "0424a15f158bbfe7277c5f84f9c55a1d4a762e60",
  "created_at": 1709327895,
  "producer": "stores/{store_hash}"
}
```

> The following price list records batch webhook events fire in response to actions that affect price list records:

Reviewer 2: ```suggestion
The following batch webhook events fire in response to actions that affect price list records:
```

| Name / Scope | Description |
|:-------------|:------------|
| `store/priceList/records/created/` | Fires when price list records are created. | 
| `store/priceList/records/updated/` | Fires when price list records are updated. | 
| `store/priceList/records/deleted/` | Fires when price list records are deleted. | 

Payload objects with the following scopes take the form that follows:
* `store/priceList/records/created`
* `store/priceList/records/updated/`
* `store/priceList/records/deleted/`

> ```json filename="Example price list records created payload object" showLineNumbers copy

Reviewer 2: ```suggestion
```json filename="Example payload object for creating price list records" showLineNumbers copy
```

{
  "scope": "store/priceList/records/created",
  "store_id": "1001197568",
  "data": [
     0: {

>     "price_list_id": 1,

Reviewer 2: ```suggestion
       "price_list_id": 1,
```

    "variant_id": 179,
    "currency": "USD"
        },
     1: {
    "price_list_id": 2,
    "variant_id": 180,
    "currency": "USD"
        },
   ],
  "hash": "0424a15f158bbfe7277c5f84f9c55a1d4a762e60",
  "created_at": 1709327895,
  "producer": "stores/{store_hash}"
}
```

## Price list assignments

> The following price list assignment webhook event fires in response to price list assignment changes:

Reviewer 2: ```suggestion
The following webhook events fire in response to price list assignment changes:
```

| Name / Scope | Description | 
|:-------------|:------------|
| `store/priceList/assignment/updated` | Fires when a price list assignment is assigned, reassigned, or unassigned. | 
| `store/priceList/assignment/deleted` | Fires when a price list assignment is deleted. | 

Price list assignment payload objects take the form that follows:
* `store/priceList/assignment/updated`
* `store/priceList/assignment/deleted`

> ```json filename="Example price list assignment payload object" showLineNumbers copy

Reviewer 2: ```suggestion
```json filename="Example payload object for price list assignment" showLineNumbers copy
```

Consult the [pages section of the Channel Webhooks Guide](/docs/integrations/webhooks/events/channels#pages).

## Price Lists
The following price lists webhook events fire in response to actions that affect a specific price list:

| Name / Scope | Description |
|:-------------|:------------|
| `store/priceList/created`  | Fires when a price list is created. | 
| `store/priceList/updated/` | Fires when a price list is updated. | 
| `store/priceList/activated/` | Fires when a price list is activated. | 
| `store/priceList/deactivated/` | Fires when a price list is deactivated. | 

> | `store/priceList/deleted/` | Fires when a price list is deleted. | 

Reviewer: There is a `store/priceLists/deleted` batch event with a data payload like ```{"price_list_ids":[234]}```

Author: I added the schema for batch delete (`store/priceLists/deleted`) although it is confusing because I could only remove one price list at a time. I tried adding values separated by a comma but that did not work. 

If you have an example of this I can add the example to the endpoint. Also I see the `stores/priceList/deleted`  does not exist.
![Screenshot 2024-03-07 at 10 33 25 AM](https://github.com/bigcommerce/docs/assets/64287776/67e34911-165d-4ca6-a12a-952fac4b080e)

Payload objects with the following scopes take the form that follows:

> * `store/pricelist/created`

Reviewer: Does case of `priceList` matter here?

Author: `store/pricelist/created` does exist.
![Screenshot 2024-03-07 at 10 31 42 AM](https://github.com/bigcommerce/docs/assets/64287776/6191a475-e4c7-4cf5-981c-974ea4d0f8df)
